addons/hr_expense/models/product_template.py

```diff
@@ -20,3 +21,9 @@ def default_get(self, fields):
     @api.depends('type')
     def _compute_can_be_expensed(self):
         self.filtered(lambda p: p.type not in ['consu', 'service']).update({'can_be_expensed': False})
+
+    def unlink(self):
+        product_data = [self.env.ref('hr_expense.product_product_fixed_cost', False), self.env.ref('hr_expense.car_travel', False), self.env.ref('hr_expense.air_ticket', False)]
+        for product in self.filtered(lambda p: p.product_variant_id in product_data):
+            raise UserError(_("You cannot delete %s as it is used in another module but you can archive it instead.", product.name))
```

Comment:
Please don't use phrase "another module", because it doesn't make any sense for user who see Odoo UI and doesn't know what is current module and what is another module